### DISCUSSION TITLE
Is it possible to get the type of arduino card with Telemetrix?

### DISCUSSION THREAD

**Discussion by jerlfan** [MrYsLab/telemetrix#87] [Q&A] [answered]
Hello,

I'm using telemetrix (it's wonderful) to implement a plugin in PyMoDAQ (https://pymodaq.cnrs.fr/en/latest/). The name of the plugin is pymodaq_plugin_arduino. Its purpose is to use an Arduino to drive actuators and/or sensors. 
I'm trying to consider all the possibilities, and in particular the type of board, as an Arduino Uno, for example, doesn't have the same number of analog inputs and the same bit resolution as a Giga R1. 

Is there a way of finding out the type of board using Telemetrix? I haven't found a way to do it.

Thank you

**Comment by MrYsLab** [comment, marked as answer]
Thanks for the question. I don't own a Giga R1, but I just tried compiling [Telemetrix4Arduino](https://github.com/MrYsLab/Telemetrix4Arduino) for the Giga R1. Two of the libraries that Telemetrix4Arduino uses, SPI and DHT, do not seem compatible with Giga R1. 

If you do not need SPI or DHT support, one can get the code to compile if the[ line to include SPI](https://github.com/MrYsLab/Telemetrix4Arduino/blob/7271269667c36c1436f99432e16da2540e2db7fd/examples/Telemetrix4Arduino/Telemetrix4Arduino.ino#L45) support and the [line to support DHT ](https://github.com/MrYsLab/Telemetrix4Arduino/blob/7271269667c36c1436f99432e16da2540e2db7fd/examples/Telemetrix4Arduino/Telemetrix4Arduino.ino#L53)are commented out.

Also, Telemetrix4Arduino only supports the Serial USB and not the WiFi.

Please let me know if you have any other questions.

**Reply by MrYsLab** [reply]
I apologize for not answering your question, "Is there a way of finding out the type of board using Telemetrix?" The Arduino sketch Telemetrix4Arduino is compiled by selecting a specific board type in the Arduino IDE. The same sketch code can be used to compile for multiple boards, such as the Uno, Leonardo, and Mega2560. 
So, the board type is known at compile time but not run time, so to answer your question, telemetrix has no easy way of retrieving the board type and reporting it.
